### PR TITLE
migrate acr cli image to mariner

### PR DESCRIPTION
**Pull request by wju-MSFT**
Migrated acr cli image from docker to mariner.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+make
 
 FROM gobuild-base AS acr-cli
 WORKDIR /go/src/github.com/Azure/acr-cli
 COPY . .
 RUN make binaries && mv bin/acr /usr/bin/acr
 
-FROM alpine:3.15
-RUN apk --update add ca-certificates
+FROM mcr.microsoft.com/cbl-mariner/base/core:2.0.20221122
```
Comment (northtyphoon) [inline Dockerfile:11]: I think we can use the stable tag `mcr.microsoft.com/cbl-mariner/base/core:2.0`

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+RUN tdnf --refresh install -y ca-certificates
```
Comment (northtyphoon) [inline Dockerfile:12]: We can use `ca-certificates-microsoft`

**Review by northtyphoon — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -1,14 +1,14 @@
-FROM golang:1.18.1-alpine3.15 AS gobuild-base
-RUN apk add --no-cache \
-	git \
-	make
+FROM mcr.microsoft.com/oss/go/microsoft/golang:1.18-fips-cbl-mariner2.0 AS gobuild-base
+RUN tdnf clean all && tdnf install -y \
```
Comment (northtyphoon) [inline Dockerfile:2]: I'm thinking we can use the following command to install packages on the container image

RUN tdnf check-update \
    && tdnf install -y \
        git \
        make \
    && tdnf clean all

File: Dockerfile (hunk 2 above)
Comment (wju-MSFT) [reply Dockerfile:12]: Done

File: Dockerfile (hunk 3 above)
Comment (wju-MSFT) [reply Dockerfile]: Reflected on both builder and runtime.

File: Dockerfile (hunk 1 above)
Comment (wju-MSFT) [reply Dockerfile:11]: Done

**Review by northtyphoon — APPROVED** [review]

**Pull request merged**